internal/shellgen/scripts.go

```diff
@@ -82,39 +81,53 @@ func WriteScriptsToFiles(devbox devboxer) error {
 	return nil
 }
 
-func WriteScriptFile(devbox devboxer, name string, body string) (err error) {
-	script, err := os.Create(ScriptPath(devbox.ProjectDir(), name))
+func writeHookFile(devbox devboxer, body string) (err error) {
+	script, err := createScriptFile(devbox, HooksFilename)
 	if err != nil {
 		return errors.WithStack(err)
 	}
-	defer func() {
-		cerr := script.Close()
-		if err == nil {
-			err = cerr
-		}
-	}()
-	err = script.Chmod(0755)
+	defer func() { _ = script.Close() }() // best effort: close file
```

Comment:
Does `defer script.Close()` not work? (and add `nolint` if you want to ignore error)